frontend/public/components/utils/kebab.tsx

```diff
@@ -218,42 +218,73 @@ export const ResourceKebab = connectToModel((props: ResourceKebabProps) => {
   />;
 });
 
-export class Kebab extends DropdownMixin {
-  static factory: KebabFactory = kebabFactory;
-
-  onClick = (event, option: KebabOption) => {
-    event.preventDefault();
-
-    if (option.callback) {
-      option.callback();
-    }
+const KebabHook: React.FC<KebabProps> = ({ options, isDisabled, position, id, children }) => {
+  const [isOpen, setIsOpen] = React.useState(false);
 
-    if (option.href) {
-      history.push(option.href);
-    }
+  function onToggle(open) {
+    setIsOpen(open);
+  }
 
-    this.hide();
-  };
+  function onSelect() {
+    isOpen ? setIsOpen(false) : setIsOpen(true);
+  }
 
-  onHover = () => {
+  function onHover() {
```

Comment:
`onHover` cannot be the only trigger to check access. This will break with keyboard / screen reader access and potentially on mobile. This was simply a performance optimization to prefetch things before the dropdown opens to avoid flicker.